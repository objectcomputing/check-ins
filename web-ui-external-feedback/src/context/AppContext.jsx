import React, { useEffect, useReducer, useMemo } from 'react';
import { reducer, initialState } from './reducer';
import {
  SET_CSRF,
} from './actions';
import { BASE_API_URL } from '../api/api';
;

const AppContext = React.createContext();

function getSessionCookieValue(name) {
  const cookies = document?.cookie?.split(';');
  for (let i = 0; i < cookies.length; i++) {
    const cookie = cookies[i].trim();
    if (cookie.startsWith(name + '=')) {
      return decodeURIComponent(cookie.substring(name.length + 1));
    }
  }
  return null;
}

const AppContextProvider = props => {
  const [state, dispatch] = useReducer(
    reducer,
    props?.value?.state || initialState
  );

  const {
    csrf,
  } = state;
  const url = `${BASE_API_URL}/csrf/cookie`;
  useEffect(() => {
    const getCsrf = async () => {
      if (!csrf) {
        const payload = getSessionCookieValue('_csrf');
        if (payload) {
          dispatch({ type: SET_CSRF, payload });
        } else {
          const res = await fetch(url, {
            responseType: 'text',
            credentials: 'include'
          });
          if (res && res.ok) {
            dispatch({ type: SET_CSRF, payload: await res.text() });
          }
        }
      }
    };
    getCsrf();
  }, [csrf]);

  const value = useMemo(() => {
    return { state, dispatch };
  }, [state]);
  return (
    <AppContext.Provider value={props.value || value}>
      {props.children}
    </AppContext.Provider>
  );
};

export { AppContext, AppContextProvider };
