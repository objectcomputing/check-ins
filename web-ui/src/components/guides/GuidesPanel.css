.guide-container {
    width: fit-content;
    height: fit-content;
    border: solid 3px black;
    margin-top: 1rem;
  }
  