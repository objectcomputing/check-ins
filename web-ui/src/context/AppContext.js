import React, { useEffect, useReducer, useMemo } from "react";
import { createSelector } from 'reselect';
import { getCurrentUser, updateMember, getAllMembers } from "../api/member";
import { getCheckinByMemberId, createCheckin } from "../api/checkins";
import { BASE_API_URL } from "../api/api";
import axios from "axios";

export const MY_PROFILE_UPDATE = "@@check-ins/update_profile";
export const UPDATE_USER_BIO = "@@check-ins/update_bio";
export const UPDATE_CHECKINS = "@@check-ins/update_checkins";
export const UPDATE_INDEX = "@@check-ins/update_index";
export const UPDATE_TOAST = "@@check-ins/update_toast";
export const UPDATE_CURRENT_CHECKIN = "@@check-ins/update_current_checkin";
export const UPDATE_TEAMS = "@@check-ins/update_teams";
export const UPDATE_MEMBER_PROFILES = "@@check-ins/update_member_profiles";
export const UPDATE_TEAM_MEMBERS = "@@check-ins/update_team_members";
export const UPDATE_SELECTED_PROFILE = "@@check-ins/update_selected_profile";
const SET_CSRF = "@@check-ins/update_csrf";

const AppContext = React.createContext();

const reducer = (state, action) => {
  switch (action.type) {
    case MY_PROFILE_UPDATE:
      state.userProfile = action.payload;
      break;
    case UPDATE_USER_BIO:
      state.userProfile.memberProfile.bioText = action.payload;
      updateMember(state.userProfile.memberProfile);
      break;
    case UPDATE_CHECKINS:
      state.checkins = action.payload;
      //sort by date
      state.checkins.sort(function (a, b) {
        return new Date(...a.checkInDate) - new Date(...b.checkInDate);
      });
      state.currentCheckin = state.checkins[state.checkins.length - 1];
      break;
    case SET_CSRF:
      state.csrf = action.payload;
      break;
    case UPDATE_TOAST:
      state.toast = action.payload;
      break;
    case UPDATE_TEAMS:
      state.teams = action.payload;
      break;
    case UPDATE_MEMBER_PROFILES:
      state.memberProfiles = action.payload;
      break;
    case UPDATE_TEAM_MEMBERS:
      state.teamMembers = action.payload;
      break;
    case UPDATE_CURRENT_CHECKIN:
      state.currentCheckin = action.payload;
      break;
    case UPDATE_SELECTED_PROFILE:
      const { payload } = action;
      if (state.selectedProfile !== payload) {
        state.selectedProfile = payload;
        state.currentCheckin = payload ? payload.checkIn : {};
      }
      if (payload === undefined) {
        state.checkins = [];
      }
      break;
    default:
  }
  return { ...state };
};

const initialState = {
  checkins: [],
  csrf: undefined,
  currentCheckin: {},
  teams: [],
  memberProfiles: [],
  index: 0,
  selectedProfile: undefined,
  toast: {
    severity: "",
    toast: "",
  },
  userProfile: undefined,
};

const getCheckins = async (id, pdlId, date, dispatch, csrf) => {
  const res = await getCheckinByMemberId(id, csrf);
  let data =
    res.payload && res.payload.data && res.payload.status === 200 && !res.error
      ? res.payload.data
      : null;
  if (data && data.length > 0) {
    const allComplete = data.every((checkin) => checkin.completed === true);
    if (allComplete) {
      const prevCheckinDate = data[data.length - 1].checkInDate;
      if (pdlId) {
        const res = await createCheckin({
          teamMemberId: id,
          pdlId: pdlId,
          checkInDate: date(3, prevCheckinDate),
          completed: false,
        });
        const checkin =
          res.payload && res.payload.data && !res.error
            ? res.payload.data
            : null;
        data.push(checkin);
      }
    }
  } else if (data && data.length === 0) {
    if (pdlId) {
      const res = await createCheckin({
        teamMemberId: id,
        pdlId: pdlId,
        checkInDate: date(1),
        completed: false,
      });
      const checkin =
        res.payload && res.payload.data && !res.error ? res.payload.data : null;
      data = [checkin];
    }
  }
  //without this check you get infinite checkin calls
  if (data.length > 0) {
    dispatch({ type: UPDATE_CHECKINS, payload: data });
  }
};

const AppContextProvider = (props) => {
  const [state, dispatch] = useReducer(reducer, initialState);
  const memberProfile =
    state && state.userProfile && state.userProfile.memberProfile
      ? state.userProfile.memberProfile
      : undefined;
  const id = memberProfile ? memberProfile.id : undefined;
  const selectedProfile = state && state.selectedProfile;
  const selectedId = selectedProfile ? selectedProfile.id : undefined;


  const { csrf } = state;

  useEffect(() => {
    const getCsrf = async () => {
      if (!csrf) {
        const res = await axios({
          url: `${BASE_API_URL}/csrf/cookie`,
          responseType: "text",
          withCredentials: true,
        });
        if (res && res.data) {
          dispatch({ type: SET_CSRF, payload: res.data });
        }
      }
    };
    getCsrf();
  }, [csrf]);

  useEffect(() => {
    const updateUserProfile = async () => {
      let res = await getCurrentUser(csrf);
      let profile =
        res.payload && res.payload.data && !res.error
          ? res.payload.data
          : undefined;

      if (profile) {
        dispatch({ type: MY_PROFILE_UPDATE, payload: profile });
      }
    };
    if (csrf) {
      updateUserProfile();
    }
  }, [csrf]);

  const date = (months, prevCheckinDate) => {
    let currentMonth = new Date().getMonth() + 1;
    let newDate = prevCheckinDate ? new Date(...prevCheckinDate) : new Date();
    if (prevCheckinDate) {
      let prevCheckinMonth = prevCheckinDate.getMonth() + 1;
      newDate.setMonth(
        newDate.getMonth() + currentMonth - prevCheckinMonth >= 3 ? 1 : months
      );
    } else {
      newDate.setMonth(newDate.getMonth() + months);
    }
    const year = newDate.getFullYear();
    const month = newDate.getMonth() + 1;
    const day = newDate.getDate();
    const hours = newDate.getHours();
    const minutes = newDate.getMinutes();
    const dateTimeArray = [year, month, day, hours, minutes, 0];
    return dateTimeArray;
  };

  useEffect(() => {
    async function getMemberProfiles() {
      let res = await getAllMembers(csrf);
      let profiles =
        res.payload && res.payload.data && !res.error
          ? res.payload.data
          : undefined;

      if (profiles) {
        dispatch({ type: UPDATE_MEMBER_PROFILES, payload: profiles });
      }
    }
    if (csrf) {
      getMemberProfiles();
    }
  }, [csrf]);

  useEffect(() => {
    if (selectedId && csrf) {
      getCheckins(selectedId, id, date, dispatch, csrf);
    }
  }, [csrf, selectedId, id]);

  const value = useMemo(() => {
    return { state, dispatch };
  }, [state]);
  return (
    <AppContext.Provider value={props.value || value}>
      {props.children}
    </AppContext.Provider>
  );
};

export const selectMemberProfiles = state => state.memberProfiles;
export const selectTeamMembers = state => state.teamMembers;

export const selectProfileMap = createSelector(
  selectMemberProfiles,
  (memberProfiles) => {
    if (memberProfiles && memberProfiles.length) {
      memberProfiles = memberProfiles.reduce((mappedById, profile) => {
        mappedById[profile.id] = profile;
        return mappedById;
      }, {});
    }
    return memberProfiles;
  }
);

export const selectMembersByTeamId = id => createSelector(
  selectTeamMembers,
  (teamMembers) => {
    let members = [];
    if (teamMembers && teamMembers.length) {
      members = teamMembers.filter((member) => member.teamid === id);
    }
    return members;
  }
);

export const selectMemberProfilesByTeamId = id => createSelector(
  selectMembersByTeamId(id),
  selectProfileMap,
  (members, profileMap) => members.map((member) => {
      return { ...profileMap[member.memberid], ...member };
    })
);

export { AppContext, AppContextProvider };
