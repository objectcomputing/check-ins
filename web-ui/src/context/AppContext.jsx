import React, { useEffect, useReducer, useMemo } from 'react';
import { reducer, initialState } from './reducer';
import { getCheckins, getAllCheckinsForAdmin } from './thunks';
import {
  MY_PROFILE_UPDATE,
  SET_CSRF,
  SET_ROLES,
  SET_USER_ROLES,
  UPDATE_GUILDS,
  UPDATE_MEMBER_SKILLS,
  UPDATE_MEMBER_PROFILES,
  UPDATE_TERMINATED_MEMBERS,
  UPDATE_SKILLS,
  UPDATE_CERTIFICATIONS,
  UPDATE_TEAMS,
  UPDATE_PEOPLE_LOADING,
  UPDATE_TEAMS_LOADING
} from './actions';
import {
  getCurrentUser,
  getAllMembers,
  getAllTerminatedMembers
} from '../api/member';
import {
  selectCanViewCheckinsPermission,
  selectCanViewTerminatedMembers,
} from './selectors';
import { getAllRoles, getAllUserRoles } from '../api/roles';
import { getMemberSkills } from '../api/memberskill';
import { BASE_API_URL } from '../api/api';
import { getAllGuilds } from '../api/guild';
import { getSkills } from '../api/skill';
import { getAllTeams } from '../api/team';
import {getCertifications} from "../api/certification.js";

const AppContext = React.createContext();

function getSessionCookieValue(name) {
  const cookies = document?.cookie?.split(';');
  for (let i = 0; i < cookies.length; i++) {
    const cookie = cookies[i].trim();
    if (cookie.startsWith(name + '=')) {
      return decodeURIComponent(cookie.substring(name.length + 1));
    }
  }
  return null;
}

const AppContextProvider = props => {
  const [state, dispatch] = useReducer(
    reducer,
    props?.value?.state || initialState
  );
  const userProfile =
    state && state.userProfile ? state.userProfile : undefined;
  const memberProfile =
    userProfile && userProfile.memberProfile
      ? userProfile.memberProfile
      : undefined;

  const id = memberProfile ? memberProfile.id : undefined;
  const pdlId = memberProfile ? memberProfile.pdlId : undefined;
  const {
    csrf,
    guilds,
    teams,
    memberSkills,
    memberProfiles,
    checkins,
    skills,
    certifications,
    roles,
    userRoles
  } = state;
  const url = `${BASE_API_URL}/csrf/cookie`;
  useEffect(() => {
    const getCsrf = async () => {
      if (!csrf) {
        const payload = getSessionCookieValue('_csrf');
        if (payload) {
          dispatch({ type: SET_CSRF, payload });
        } else {
          const res = await fetch(url, {
            responseType: 'text',
            credentials: 'include'
          });
          if (res && res.ok) {
            dispatch({ type: SET_CSRF, payload: await res.text() });
          }
        }
      }
    };
    getCsrf();
  }, [csrf]);

  useEffect(() => {
    async function getGuilds() {
      let res = await getAllGuilds(csrf);
      let data =
        res.payload &&
        res.payload.data &&
        res.payload.status === 200 &&
        !res.error
          ? res.payload.data
          : null;
      if (data) {
        dispatch({ type: UPDATE_GUILDS, payload: data });
      }
    }
    if (csrf && !guilds) {
      getGuilds();
    }
  }, [csrf, guilds]);

  useEffect(() => {
    async function getTeams() {
      let res = await getAllTeams(csrf);
      let data =
        res.payload &&
        res.payload.data &&
        res.payload.status === 200 &&
        !res.error
          ? res.payload.data
          : null;
      if (data) {
        dispatch({ type: UPDATE_TEAMS, payload: data });
        dispatch({ type: UPDATE_TEAMS_LOADING });
      }
    }
    if (csrf && !teams) {
      dispatch({ type: UPDATE_TEAMS_LOADING });
      getTeams();
    }
  }, [csrf, teams, dispatch]);

  useEffect(() => {
    const updateUserProfile = async () => {
      let res = await getCurrentUser(csrf);
      let profile =
        res.payload && res.payload.data && !res.error
          ? res.payload.data
          : undefined;

      if (profile) {
        dispatch({ type: MY_PROFILE_UPDATE, payload: profile });
      }
    };
    if (csrf && !userProfile) {
      updateUserProfile();
    }
  }, [csrf, userProfile]);

  useEffect(() => {
    const getAllMemberSkills = async () => {
      const res = await getMemberSkills(csrf);
      const memberSkills =
        res && res.payload && res.payload.data ? res.payload.data : null;
      if (memberSkills) {
        dispatch({ type: UPDATE_MEMBER_SKILLS, payload: memberSkills });
      }
    };
    if (csrf && !memberSkills) {
      getAllMemberSkills();
    }
  }, [csrf, memberSkills]);

  useEffect(() => {
    async function getMemberProfiles() {
      let res = await getAllMembers(csrf);
      let profiles =
        res.payload && res.payload.data && !res.error
          ? res.payload.data
          : undefined;

      if (profiles) {
        dispatch({ type: UPDATE_MEMBER_PROFILES, payload: profiles });
        dispatch({ type: UPDATE_PEOPLE_LOADING, payload: false });
      }
    }
    async function getTerminatedMembers() {
      let res = await getAllTerminatedMembers(csrf);
      let profiles =
        res.payload && res.payload.data && !res.error
          ? res.payload.data
          : undefined;

      if (profiles) {
        dispatch({ type: UPDATE_TERMINATED_MEMBERS, payload: profiles });
      }
    }
    if (csrf && userProfile && !memberProfiles) {
      dispatch({ type: UPDATE_PEOPLE_LOADING, payload: true });
      getMemberProfiles();
      if (selectCanViewTerminatedMembers(state)) {
        getTerminatedMembers();
      }
    }
  }, [csrf, userProfile, memberProfiles]);

  useEffect(() => {
    function getAllTheCheckins() {
      if (
        userProfile &&
        userProfile.permissions?.some(p =>
          p?.permission?.includes('CAN_VIEW_CHECKINS_REPORT')
        ) &&
        id &&
        csrf
      ) {
        getAllCheckinsForAdmin(dispatch, csrf);
      } else if (id && csrf && selectCanViewCheckinsPermission(state)) {
        getCheckins(id, pdlId, dispatch, csrf);
      }
    }
    if (csrf && !checkins) {
      getAllTheCheckins();
    }
  }, [csrf, pdlId, id, userProfile, checkins]);

  useEffect(() => {
    const getAllSkills = async () => {
      const res = await getSkills(csrf);
      const data =
        res &&
        res.payload &&
        res.payload.data &&
        res.payload.status === 200 &&
        !res.error
          ? res.payload.data
          : null;
      if (data && data.length > 0) {
        dispatch({ type: UPDATE_SKILLS, payload: data });
      }
    };
    if (csrf && !skills) {
      getAllSkills();
    }
  }, [csrf, skills]);

  useEffect(() => {
    const getAllCertifications = async () => {
      const res = await getCertifications(csrf);
      const data =
          res &&
          res.payload &&
          res.payload.data &&
          res.payload.status === 200 &&
          !res.error
              ? res.payload.data
              : null;
      if (data && data.length > 0) {
        dispatch({ type: UPDATE_CERTIFICATIONS, payload: data });
      }
    };
    if (csrf && !certifications) {
      getAllCertifications();
    }
  }, [csrf, certifications]);

  useEffect(() => {
    const getRoles = async () => {
      const res = await getAllRoles(csrf);
      const data =
        res &&
        res.payload &&
        res.payload.data &&
        res.payload.status === 200 &&
        !res.error
          ? res.payload.data
          : null;
      if (data && Array.isArray(data) && data.length > 0) {
        dispatch({ type: SET_ROLES, payload: data });
      }
    };
    if (csrf && !roles) {
      getRoles();
    }
  }, [csrf, roles]);

  useEffect(() => {
    const getUserRoles = async () => {
      // make call to the API
      let res = await getAllUserRoles(csrf);
      return res.payload &&
        res.payload.data &&
        res.payload.status === 200 &&
        !res.error
        ? res.payload.data
        : null;
    };

    if (csrf && !userRoles) {
      getUserRoles().then(userRoles => {
        dispatch({ type: SET_USER_ROLES, payload: userRoles });
      });
    }
  }, [csrf, userRoles]);

  const value = useMemo(() => {
    return { state, dispatch };
  }, [state]);
  return (
    <AppContext.Provider value={props.value || value}>
      {props.children}
    </AppContext.Provider>
  );
};

export { AppContext, AppContextProvider };
