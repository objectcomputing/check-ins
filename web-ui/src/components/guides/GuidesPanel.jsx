import React, { useContext } from "react";
import { AppContext } from "../../context/AppContext";
import "./GuidesPanel.css";
import GuideLink from "./GuideLink";

const GuidesPanel = () => {
  const { state } = useContext(AppContext);
  const { userProfile } = state;
  const isPdl =
    userProfile &&
    userProfile.role &&
    userProfile.role.length > 0 &&
    userProfile.role.includes("PDL");

  const teamMemberPDFs = [
    {
      name: "Expectations Discussion Guide for Team Members",
    },
    {
      name: "Expectations Worksheet",
    },
    {
      name: "Feedback Discussion Guide for Team Members",
    },
    {
      name: "Development Discussion Guide for Team Members",
    },
    {
      name: "Individual Development Plan",
    },
  ];

  const pdlPDFs = [
    {
      name: "Development Discussion Guide for PDLs",
    },
    {
      name: "Expectations Discussion Guide for PDLs",
    },
    {
      name: "Feedback Discussion Guide for PDLs",
    },
  ];

  return (
    <fieldset className="guide-container">
      <legend>Check-In Guides</legend>
      <div>
        {teamMemberPDFs.map((memberPDF) => (
          <GuideLink key={memberPDF.name} name={memberPDF.name} />
        ))}
        {isPdl &&
          pdlPDFs.map((pdlPDF) => (
            <GuideLink key={pdlPDF.name} name={pdlPDF.name} />
          ))}
      </div>
    </fieldset>
  );
};

export default GuidesPanel;
