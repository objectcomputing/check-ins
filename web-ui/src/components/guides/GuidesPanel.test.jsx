import React from 'react';
import { MemoryRouter } from 'react-router-dom';
import GuidesPanel from './GuidesPanel';
import { AppContextProvider } from '../../context/AppContext.jsx';

const initialState = {
  state: {
    csrf: 'O_3eLX2-e05qpS_yOeg1ZVAs9nDhspEi',
    userProfile: {
      name: 'holmes',
      role: ['MEMBER'],
      imageUrl:
        'https://upload.wikimedia.org/wikipedia/commons/7/74/SNL_MrBill_Doll.jpg'
    },
    roles: [
      { id: 1, role: 'MEMBER' },
      { id: 2, role: 'PDL' }
    ],
    mockuments: [
      {
        id: 'mockument-1',
        name: 'Expectations Discussion Guide for Team Members',
        url: '/pdfs/Expectations_Discussion_Guide_for_Team_Members.pdf',
        description: 'My description'
      },
      {
        id: 'mockument-2',
        name: 'Expectations Worksheet',
        url: '/pdfs/Expectations_Worksheet.pdf'
      }
    ],
    teams: []
  }
};


it('renders correctly', () => {
  snapshot(
    <MemoryRouter>
      <AppContextProvider value={initialState}>
        <GuidesPanel />
      </AppContextProvider>
    </MemoryRouter>
  );
});
