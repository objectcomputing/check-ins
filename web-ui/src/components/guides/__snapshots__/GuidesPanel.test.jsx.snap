// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`renders correctly 1`] = `
<div>
  <div
    class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 MuiCard-root css-bhp9pd-MuiPaper-root-MuiCard-root"
  >
    <div
      class="MuiCardHeader-root css-185gdzj-MuiCardHeader-root"
    >
      <div
        class="MuiCardHeader-avatar css-1ssile9-MuiCardHeader-avatar"
      >
        <svg
          aria-hidden="true"
          class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
          data-testid="PictureAsPdfIcon"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M20 2H8c-1.1 0-2 .9-2 2v12c0 1.1.9 2 2 2h12c1.1 0 2-.9 2-2V4c0-1.1-.9-2-2-2m-8.5 7.5c0 .83-.67 1.5-1.5 1.5H9v2H7.5V7H10c.83 0 1.5.67 1.5 1.5zm5 2c0 .83-.67 1.5-1.5 1.5h-2.5V7H15c.83 0 1.5.67 1.5 1.5zm4-3H19v1h1.5V11H19v2h-1.5V7h3zM9 9.5h1v-1H9zM4 6H2v14c0 1.1.9 2 2 2h14v-2H4zm10 5.5h1v-3h-1z"
          />
        </svg>
      </div>
      <div
        class="MuiCardHeader-content css-1qbkelo-MuiCardHeader-content"
      >
        <span
          class="MuiTypography-root MuiTypography-body2 MuiCardHeader-title css-et1ao3-MuiTypography-root"
        >
          Team Member Resources
        </span>
      </div>
    </div>
    <ul
      class="MuiList-root MuiList-padding MuiList-dense css-h4y409-MuiList-root"
    >
      <a
        class="MuiListItem-root MuiListItem-dense MuiListItem-gutters MuiListItem-padding css-6lnq02-MuiListItem-root"
        href="/pdfs/Expectations_Discussion_Guide_for_Team_Members.pdf"
        target="_blank"
      >
        <div
          class="MuiListItemText-root MuiListItemText-dense MuiListItemText-multiline css-konndc-MuiListItemText-root"
        >
          <span
            class="MuiTypography-root MuiTypography-body2 MuiListItemText-primary css-et1ao3-MuiTypography-root"
          >
            Expectations Discussion Guide for Team Members
          </span>
          <p
            class="MuiTypography-root MuiTypography-body2 MuiListItemText-secondary css-83ijpv-MuiTypography-root"
          >
            My description
          </p>
        </div>
      </a>
      <a
        class="MuiListItem-root MuiListItem-dense MuiListItem-gutters MuiListItem-padding css-6lnq02-MuiListItem-root"
        href="/pdfs/Expectations_Worksheet.pdf"
        target="_blank"
      >
        <div
          class="MuiListItemText-root MuiListItemText-dense css-tlelie-MuiListItemText-root"
        >
          <span
            class="MuiTypography-root MuiTypography-body2 MuiListItemText-primary css-et1ao3-MuiTypography-root"
          >
            Expectations Worksheet
          </span>
        </div>
      </a>
    </ul>
  </div>
</div>
`;
